host.py

```diff
@@ -319,19 +319,23 @@ def run(
         *,
         env_extra: Optional[Mapping[str, Optional[str]]] = None,
         cwd: Optional[str] = None,
+        sudo: Optional[bool] = None,
     ) -> Result:
-        if self.sudo_needed and env is not None:
+        if sudo is None:
```

Comment:
Need to be very careful here. `sudo_needed` is set elsewhere. I like the fact that your code makes this less side-effecty though.